-1 
Tested at: bf7130fa368d516294111151734e7829b7a8d300
 I found a compilation error while trying to compile with clang: 
I used this command: 
 scram b vclean && scram build -k -j 24 USER_CXXFLAGS='-fsyntax-only' COMPILER='llvm compile'


 <pre>warning: unknown warning option '-Wno-psabi' [-Wunknown-warning-option]
warning: unknown warning option '-Wno-psabi' [-Wunknown-warning-option]
In file included from RecoEgammaElectronIdentification/a/RecoEgammaElectronIdentification_xr.cc:41:
In file included from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-29-2300/src/RecoEgamma/ElectronIdentification/src/classes.h:4:
In file included from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-29-2300/src/RecoEgamma/ElectronIdentification/interface/VersionedGsfElectronSelector.h:6:
/build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-29-2300/src/PhysicsTools/SelectorUtils/interface/VersionedSelector.h:104:31: error: typename is allowed for identifiers only
  using typename Selector<T>::operator();
        ~~~~~~~~~             ^
In file included from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-29-2300/src/RecoEgamma/ElectronIdentification/src/ClassBasedElectronID.cc:1:
In file included from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-29-2300/src/RecoEgamma/ElectronIdentification/interface/ClassBasedElectronID.h:4:
In file included from /build/cmsbuild/jenkins-workarea/workspace/ib-any-integration/CMSSW_7_5_X_2015-03-29-2300/src/RecoEgamma/ElectronIdentification/interface/ElectronIDAlgo.h:14:
</pre> 
 you can see the results of the tests here: 
 https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-8493/3922/summary.html

--- a/DataFormats/PatCandidates/interface/Photon.h
+++ b/DataFormats/PatCandidates/interface/Photon.h
@@ -135,16 +135,16 @@ namespace pat {
       float patParticleIso() const { return userIsolation(pat::PfAllParticleIso); }
       /// Returns the isolation calculated with only the charged hadron
       /// PFCandidates
-      float patChargedHadronIso() const { return userIsolation(pat::PfChargedHadronIso); }
+      float chargedHadronIso() const { return reco::Photon::chargedHadronIso(); }
       /// Returns the isolation calculated with only the neutral hadron
       /// PFCandidates
-      float patNeutralHadronIso() const { return userIsolation(pat::PfNeutralHadronIso); }        
+      float neutralHadronIso() const { return reco::Photon::neutralHadronIso(); }
       /// Returns the isolation calculated with only the gamma
       /// PFCandidates
-      float patPhotonIso() const { return userIsolation(pat::PfGammaIso); }
+      float photonIso() const { return reco::Photon::photonIso(); }
       /// Returns the isolation calculated with only the pile-up charged hadron
       /// PFCandidates
-      float patPuChargedHadronIso() const { return userIsolation(pat::PfPUChargedHadronIso); }        
+      float puChargedHadronIso() const { return userIsolation(pat::PfPUChargedHadronIso); }        
 
       /// Returns a user defined isolation value
       float userIso(uint8_t index=0)  const { return userIsolation(IsolationKeys(UserBaseIso + index)); }

--- a/PhysicsTools/SelectorUtils/interface/VersionedSelector.h
+++ b/PhysicsTools/SelectorUtils/interface/VersionedSelector.h
@@ -101,7 +101,7 @@ class VersionedSelector : public Selector<T> {
   }
   
 #ifndef __ROOTCLING__
-  using typename Selector<T>::operator();
+  using Selector<T>::operator();
 #endif
   
   const unsigned char* md55Raw() const { return id_md5_; } 

